File: solutions/3334_apple-redistribution-into-boxes.rb
# Problem: https://leetcode.com/problems/apple-redistribution-into-boxes
# Runtime: 0 ms

# @param {Integer[]} apple
# @param {Integer[]} capacity
# @return {Integer}
def minimum_boxes(apple, capacity)
    sorted_capacity = capacity.sort.reverse
    idx = 0
    curr = apple.sum
    while curr > 0 do
        curr -= sorted_capacity[idx]
        idx += 1
    end
    return idx
end

File: solutions/3334_apple-redistribution-into-boxes.py
# Problem: https://leetcode.com/problems/apple-redistribution-into-boxes
# Runtime: 3 ms

class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        capacity.sort(reverse=True)
        total = sum(apple)
        idx = 0
        while total > 0:
            total -= capacity[idx]
            idx += 1
        return idx